packages/react-native-media-player/android/src/main/java/com/yajananrao/mediaplayer/MediaPlayerService.java

```diff
@@ -0,0 +1,594 @@
+                        Toast.makeText(getApplicationContext(), "No Internet Connection", Toast.LENGTH_SHORT).show();
+                        setMediaPlaybackState(PlaybackStateCompat.STATE_PAUSED);
+                        showPausedNotification();
+                    }
+                } else {
+                    Log.i(TAG, "waits for prepare");
+                    mMediaPlayer.prepare();
+                    play();
+                }
+            } else {
+                play();
+            }
+        } catch (Exception e) {
+            // TODO: handle exception
+            Log.e(TAG, "onPlay: Parent " + e.toString());
+        }
+    }
+
+    private void pausePlayback() {
+        if (mMediaPlayer.isPlaying()) {
+            Log.d(TAG, "on pause");
+            mMediaPlayer.pause();
+            AudioManager am = (AudioManager) getSystemService(Context.AUDIO_SERVICE);
+            setMediaPlaybackState(PlaybackStateCompat.STATE_PAUSED);
+            unregisterReceiver(mNoisyReceiver);
+            showPausedNotification();
+        }
+    }
+
+    private void stopPlayback() {
+        try {
+            AudioManager am = (AudioManager) getSystemService(Context.AUDIO_SERVICE);
+            // Abandon audio focus
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                am.abandonAudioFocusRequest(focus);
+            }
+            setMediaPlaybackState(PlaybackStateCompat.STATE_STOPPED);
+            try {
+                if (mNoisyReceiver != null) {
+                    unregisterReceiver(mNoisyReceiver);
+                }
+            } catch (IllegalArgumentException e) {
+                Log.e(TAG, "noisy receiver " + e.toString());
+            }
+
+            // walk around for notification clear issue
+            if (mMediaSessionCompat.isActive()) {
+                String channel = null;
+
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                    channel = NotificationChannel.DEFAULT_CHANNEL_ID;
+                }
+                startForeground(NOTIFICATION_ID, new NotificationCompat.Builder(getApplicationContext(), CHANNEL_ID).build());
+                stopSelf();
+            }
+            clearNotification();
+
+            // Stop the service
+            stopSelf();
+            // Set the session inactive (and update metadata and state)
+            mMediaSessionCompat.setActive(false);
+            // stop the player (custom call)
+            mMediaPlayer.stop();
+            mMediaPlayer.release();
+            // Take the service out of the foreground
+            stopForeground(true);
+        } catch (Exception e) {
+            Log.e(TAG, "onStop: " + e.toString());
+        }
+    }
+
+    private void setDataSource(final String uri) {
+        Log.d(TAG, "onPlayFromUri: song received " + uri);
+        try {
+            initalLoad = true;
+            if (mMediaPlayer != null) {
+                if (mMediaPlayer.isPlaying()) {
```

Comment:
Codacy found an issue: [These nested if statements could be combined](https://app.codacy.com/manual/yajananrao/Nadha/pullRequest?prid=5728824)